added entry for obsidian-count-down

hamolicious opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/hamolicious/Obsidian-Count-Down

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [x]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[import moment from "moment";](https://github.com/hamolicious/Obsidian-Count-Down/blob/fc2341d73c1ba871129b69b905cf4157a3f6942d/main.ts#L8)
You don't need to bring your own copy of moment, Obsidian already provides it.

[declare global](https://github.com/hamolicious/Obsidian-Count-Down/blob/fc2341d73c1ba871129b69b905cf4157a3f6942d/main.ts#L14)
Please don't do this, declaring global variables can lead to all kinds of problems.
You can just remove it, and use `this.settings` in the one place where you access it.

[this.registerMarkdownPostProcessor(async function (](https://github.com/hamolicious/Obsidian-Count-Down/blob/fc2341d73c1ba871129b69b905cf4157a3f6942d/main.ts#L31-L37)
You can just do `this.registerMarkdownPostProcessor(countdownMarkdownPostProcessor);`

[// NOTE: figure out the way to add highlighting blocks](https://github.com/hamolicious/Obsidian-Count-Down/blob/fc2341d73c1ba871129b69b905cf4157a3f6942d/main.ts#L66)
```ts
el.createEl("mark", {cls: "countdown-highlight", text: formatDiff(diff)});
```
would be my way to do this.

joethei commented on community-plugins.json:
```diff
@@ -4545,6 +4545,12 @@
+        "description": "Adds a count down to selected dates for easier timing and tracking of TODOs or events",
+        "repo": "hamolicious/Obsidian-Count-Down"
+    },
         "id": "obsidian-attendance",
```
This is not valid JSON.
there is a { missing before this.

hamolicious replied:
fixed ✅

hamolicious commented:
> import moment from "moment";
You don't need to bring your own copy of moment, Obsidian already provides it.

fixed as of [d237aaf](https://github.com/hamolicious/Obsidian-Count-Down/commit/d237aaf169e3d57fee6ff2b8f747f68b76575328)

---

> declare global
Please don't do this, declaring global variables can lead to all kinds of problems.
You can just remove it, and use this.settings in the one place where you access it.

fixed as of [b11a497](https://github.com/hamolicious/Obsidian-Count-Down/commit/b11a497e0effcd0fe5aa4c0856f1ad934a4eaa7c)

---

> this.registerMarkdownPostProcessor(async function (
You can just do this.registerMarkdownPostProcessor(countdownMarkdownPostProcessor);

fixed as of [1baf694](https://github.com/hamolicious/Obsidian-Count-Down/commit/1baf694e344de13616b0cd9578a23518929e2329)

---

> // NOTE: figure out the way to add highlighting blocks
el.createEl("mark", {cls: "countdown-highlight", text: formatDiff(diff)});
would be my way to do this.

duly noted, this is not yet a priority but thank you very much.
fixed as of [not-yet](#)

hamolicious commented:
> #### Hello hamolicious! [<img alt="" height="30px" src="https://user-images.githubusercontent.com/59741989/139557624-63e6e31f-e617-4041-89ae-78b534a8de5c.png">](https://obsidian.md)
> **I found the following errors in your plugin, Focus and Highlight:**
> 
> ❌ The newly added entry is not at the end, or you are submitting on someone else's behalf. Last plugin in the list is: nagi1999a/obsidian-focus-plugin
> 
> This check was done automatically.

this will be fixed when I am ready to merge

liamcain commented:
Hi, you might want to mention in the README that this transformation will only happen in Reading mode and not in live preview.

- [if (!doesContainDate(](https://github.com/hamolicious/Obsidian-Count-Down/blob/b11a497e0effcd0fe5aa4c0856f1ad934a4eaa7c/main.ts#L51) nit: formatting here looks wrong. Looks like the indentation is off.
- [if (text[i] != ':') continue;](https://github.com/hamolicious/Obsidian-Count-Down/blob/b11a497e0effcd0fe5aa4c0856f1ad934a4eaa7c/main.ts#L56-L57) should be `!==`
- [.fromNow()](https://github.com/hamolicious/Obsidian-Count-Down/blob/b11a497e0effcd0fe5aa4c0856f1ad934a4eaa7c/main.ts#L65-L66) you can change this to `fromNow(true)` and it will remove the "in " and " ago" parts of the string format. Then you can remove the `replace("in ", "")` line.
- [el?.innerText](https://github.com/hamolicious/Obsidian-Count-Down/blob/b11a497e0effcd0fe5aa4c0856f1ad934a4eaa7c/main.ts#L54) No need for the `?` everywhere since you have already checked that `el` is not null
- [for (let i = 0; i < el.innerText.length; i++) {](https://github.com/hamolicious/Obsidian-Count-Down/blob/b11a497e0effcd0fe5aa4c0856f1ad934a4eaa7c/main.ts#L53) This is a very inefficient way to find matches. I think you should take a look at how the Emoji Shortcodes plugin handles this (it's doing a very similar thing here: <https://github.com/phibr0/obsidian-emoji-shortcodes/blob/83a9e4990bc8a87ff14a98ba0b54caa4426c6732/src/emojiPostProcessor.ts>. You're basic process should be searching for the date (the same way you are in `doesContainDate`, then replacing that match with the moment string you generated.
- [null != text.match(/(:[0-9][0-9]\/[0-9][0-9]\/[0-9][0-9][0-9][0-9]:)/)](https://github.com/hamolicious/Obsidian-Count-Down/blob/b11a497e0effcd0fe5aa4c0856f1ad934a4eaa7c/main.ts#L86-L87) You are hard coding the date format here. But you are also providing a setting for users do specify their own date format. This won't work.

ericaxu commented:
@hamolicious Hi! Do you still intend to submit this plugin? Thanks!

ericaxu commented:
> > @hamolicious Hi! Do you still intend to submit this plugin? Thanks!
> 
> No, sorry forgot to close, I need to iron out some bugs first and need to find the time to do that, will resubmit when I get a chance

Great, thanks for the update!
